scalacheck/src/main/scala/pathy/scalacheck/RandomStr.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2014 - 2015 SlamData Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pathy
+package scalacheck
+
+import org.scalacheck.Arbitrary
+import org.scalacheck.Gen
+
+import scalaz.Show
+
+final case class RandomStr(str: String) extends AnyVal
```

Comment:
The name `RandomStr` still irritates me a bit, but I am cool with it as long as we make this class private to this package. It's an implementation detail of how we supply an `Arbitrary[Path[B,T,S]]` as opposed to a `ConstrainedPath[B,T,S,A]` without too much code duplication.